homeassistant/components/picnic/todo.py

```diff
@@ -27,19 +33,20 @@ async def async_setup_entry(
     """Set up the Picnic shopping cart todo platform config entry."""
     picnic_coordinator = hass.data[DOMAIN][config_entry.entry_id][CONF_COORDINATOR]
 
-    # Add an entity shopping card
-    async_add_entities([PicnicCart(picnic_coordinator, config_entry)])
+    async_add_entities([PicnicCart(hass, picnic_coordinator, config_entry)])
 
 
 class PicnicCart(TodoListEntity, CoordinatorEntity):
     """A Picnic Shopping Cart TodoListEntity."""
 
     _attr_has_entity_name = True
-    _attr_translation_key = "shopping_cart"
     _attr_icon = "mdi:cart"
+    _attr_supported_features = TodoListEntityFeature.CREATE_TODO_ITEM
+    _attr_translation_key = "shopping_cart"
 
     def __init__(
         self,
+        hass: HomeAssistant,
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.